### PR TITLE
feat[Gameplay]: Add Allura behaviour

### PR DESCRIPTION
**Pull request by DADSILVER**
You can test all this things in Gameplay_MiniBoss2, Gameplay_MiniBoss2_with_allura and __LEVEL1_ALFA
-Add new character.
-Fix state machines
-Change BixAttack script to Player Attackscript

### CODE REVIEW

File: Source/DataModels/Components/ComponentTransform.cpp [hunk 1]
```diff
@@ -69,10 +69,10 @@ void ComponentTransform::InternalSave(Json& meta)
 	meta["localSca_Z"] = static_cast<float>(localSca.z);
 
 	//Use this when you want to save ANY old scene with new boundingBox Value
-	//if (bbSca.x == 0.0f && bbSca.y == 0.0f && bbSca.z == 0.0f)
-	//{
-	//	bbSca = { 1.0f,1.0f,1.0f };
-	//}
+	if (bbSca.x == 0.0f && bbSca.y == 0.0f && bbSca.z == 0.0f)
```
Comment (AuxPort66) [inline Source/DataModels/Components/ComponentTransform.cpp:72]: Comment this again pls

File: Game/Scripts/PlayerMoveScript.cpp [hunk 2]
```diff
@@ -41,10 +41,13 @@ void PlayerMoveScript::Start()
...
 	playerManager = owner->GetComponent<PlayerManagerScript>();
-	forceScript = owner->GetComponent<PlayerForceUseScript>();
+	if (owner->HasComponent<PlayerForceUseScript>())
+	{
+		forceScript = owner->GetComponent<PlayerForceUseScript>();
+	}
 	rigidBody = owner->GetComponent<ComponentRigidBody>();
 	jumpScript = owner->GetComponent<PlayerJumpScript>();
-	bixAttackScript = owner->GetComponent<BixAttackScript>();
+	bixAttackScript = owner->GetComponent<PlayerAttackScript>();
```
Comment (UriMurcia) [inline Game/Scripts/PlayerMoveScript.cpp:50]: Missing bixAttackScript name changed

File: Game/Scripts/RangedFastAttackBullet.cpp [hunk 3]
```diff
@@ -54,7 +55,7 @@ void RangedFastAttackBullet::OnCollisionEnter(ComponentRigidBody* other)
 		return;
 	}
 
-	if (other->GetOwner()->CompareTag("Player"))
+	if (other->GetOwner()->CompareTag(targetTag))
```
Comment (UriMurcia) [inline Game/Scripts/RangedFastAttackBullet.cpp:58]: Why did you change this? Was not okay before? I see that you only use it at one script

**Review by UriMurcia — COMMENTED** [review]
Hi David, from coding everything seems to look good, just two things:

File: Source/DataModels/Components/ComponentTransform.cpp (hunk 1 above)
Comment (DADSILVER) [reply Source/DataModels/Components/ComponentTransform.cpp]: Done

File: Game/Scripts/PlayerMoveScript.cpp (hunk 2 above)
Comment (DADSILVER) [reply Game/Scripts/PlayerMoveScript.cpp]: Done

File: Game/Scripts/RangedFastAttackBullet.cpp (hunk 3 above)
Comment (DADSILVER) [reply Game/Scripts/RangedFastAttackBullet.cpp:58]: I changed because my first idea was to modularize this script and use it as a bullet. But the bullet of the light attack works better. So I keep this implementation because maybe, in the future, we need that other thing (spaceship, a new ally, etc...) to shoot this kind of bullet. But before it worked fine.

**Comment by UriMurcia** [comment]
There are some problems with the gameplay:

- Sounds used are the same as bix, it's not your fault cause in Unity we would be able to change the sound from the inspector, without depending on the script used. But we need to see what we do.
- Light finisher seems like is shooting a bullet? It should throw an electric grenade to stun the enemies.
- Heavy finisher is throwing the lightsaber of bix, it should throw a projectile/misile.
- Does Allura not have double jump?
- Dash animation should be a roll?
- Scene Miniboss2_withAllura still crashes, it's missing the target in BoostOfEnergy object, enemies do not have a target neither, and maybe more errors. (Haven't tried the other scenes)

File: Game/Scripts/HealthSystem.cpp [hunk 4]
```diff
@@ -51,30 +51,30 @@ void HealthSystem::Start()
 		maxHealth = currentHealth;
 	}
 
-	meshEffect->FillMeshes(owner);
-	meshEffect->ReserveSpace(1);
-	meshEffect->AddColor(float3(1.f, 0.f, 0.f));
+	//meshEffect->FillMeshes(owner);
```
[the remark this answers is not in the document]
Comment (gerard-queralt) [reply Game/Scripts/HealthSystem.cpp:54]: Chiming in to remind you you can use `AXO_TODO` to add your messages in the compilation so you don't forget 😉

File: Game/Scripts/HeavyFinisherAttack.cpp [hunk 5]
```diff
@@ -168,6 +170,14 @@ void HeavyFinisherAttack::MoveForward(float deltaTime)
...
 	target->GetOwner()->GetComponent<HealthSystem>()->TakeDamage(damage);
+	if (playerAttackScript->IsMeleeAvailable())
+	{
+		audioSource->PostEvent(AUDIO::SFX::PLAYER::WEAPON::LIGHTSABER_CLASH);
+	}
+	else
+	{
+		audioSource->PostEvent(AUDIO::SFX::PLAYER::WEAPON::CANNON_SHOT_IMPACT); // Provisional sfx
+	}
 	audioSource->PostEvent(AUDIO::SFX::PLAYER::WEAPON::LIGHTSABER_CLASH);
```
Comment (UriMurcia) [inline Game/Scripts/HeavyFinisherAttack.cpp:181]: This line should be deleted.

File: Game/Scripts/HealthSystem.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (UriMurcia) [reply Game/Scripts/HealthSystem.cpp:54]: This is already functional, you just have to add the script MeshEffect to all the enemies and the player. I'd also suggest you delete the RegisterField of this variable and you write a GetComponent in the Start of the script so you don't have to put them all manually

**Review by UriMurcia — COMMENTED** [review]
Amazing job David, Allura feels amazing! Just two things

**Comment by UriMurcia** [comment]
Also please remember to set the target variable in the BoostOfEnergy object in the MiniBoss (target should be Allura)

**Review by UriMurcia — APPROVED** [review]

**Review by diegusmejia117 — APPROVED** [review]
